src/device/vi/vi_controller.c

```diff
@@ -89,13 +85,19 @@ int read_vi_regs(void* opaque, uint32_t address, uint32_t* value)
     {
         /* XXX: update current line number */
         cp0_update_count(vi->r4300);
-        if (vi->alternate_timing)
-            vi->regs[VI_CURRENT_REG] = (vi->delay - (vi->next_vi - cp0_regs[CP0_COUNT_REG])) % (NTSC_VERTICAL_RESOLUTION + 1);
+        int remaining_vi_duration = vi->next_vi - cp0_regs[CP0_COUNT_REG];
+
+        if (remaining_vi_duration < 40)
```

Comment:
Naming this magic const and adding a comment like in cen64 would help understanding this snippet.